lib/settings.js

```diff
@@ -333,34 +186,14 @@ class Settings extends EventEmitter {
    * @param {Object} opts
    * @private
    */
-  _deleteValueAtKeyPath(keyPath, opts) {
+  _deleteValueAtKeyPath(keyPath) {
     if (keyPath === '') {
-      this._writeSettings({}, opts);
+      this._cacheObj = {};
     } else {
-      const obj = this._readSettings();
-      const exists = Helpers.hasKeyPath(obj, keyPath);
-
-      if (exists) {
-        Helpers.deleteValueAtKeyPath(obj, keyPath);
-        this._writeSettings(obj, opts);
-      }
+      this._cacheObj = _.omit(this._cacheObj, keyPath);
```

Comment:
I don't think we need omit here since it creates a new object, we should be using _.unset https://lodash.com/docs/4.17.5#unset

```
_.unset(this._cacheObj, keyPath)
```